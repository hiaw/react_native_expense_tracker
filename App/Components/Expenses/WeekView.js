import React, { Component } from 'react'
import { View, Text, ScrollView, Button } from 'react-native'

import ExpenseRowView from './ExpenseRowView.js'

import styles from './ExpensesList.style.js'

export default class WeekView extends Component {

  print () {
    console.log('printing ')
  }

  _renderExpense (expense) {
    return (
      <ExpenseRowView key={expense._id} expense={expense} />
    )
  }

  render () {
    let expenses = this.props.data.map(this._renderExpense.bind(this))
    let total = this.props.data.reduce((acc, val) => {
      return acc + val.amount
    }, 0)
    let average = Math.round(total * 100 / 7) / 100
    return (
      <View style={styles.container}>
        <View style={{backgroundColor: 'lightblue'}}>
          <Text key='week' style={styles.total}>Week {this.props.week}</Text>
          <Text key='total' style={styles.total}>Total Amount: ${total}</Text>
          <Text key='average' style={styles.total}>Average Day Spending: ${average}</Text>
        </View>
        <ScrollView>
          {expenses}
        </ScrollView>
        <View style={{backgroundColor: 'lightblue'}}>
          <Button title='Print' onPress={this.print.bind(this)} />
        </View>
      </View>
    )
  }
}
