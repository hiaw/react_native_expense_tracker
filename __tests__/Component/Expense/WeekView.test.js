import 'react-native'
import React from 'react'
import WeekView from '../../../App/Components/Expenses/WeekView.js'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer'

let data = [{
  _id: 'adsflkja',
  date: new Date(2000, 1, 1),
  description: 'description',
  amount: 99.00,
  comment: 'Some long comment'
}]

it('renders correctly', () => {
  const tree = renderer.create(
    <WeekView data={data} week={4} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
