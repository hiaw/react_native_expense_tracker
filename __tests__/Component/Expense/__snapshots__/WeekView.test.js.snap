// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "marginTop": 65,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "lightblue",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "grey",
          "fontSize": 20,
          "marginLeft": 10,
        }
      }
    >
      Week 
      4
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "grey",
          "fontSize": 20,
          "marginLeft": 10,
        }
      }
    >
      Total Amount: $
      99
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "grey",
          "fontSize": 20,
          "marginLeft": 10,
        }
      }
    >
      Average Day Spending: $
      14.14
    </Text>
  </View>
  <RCTScrollView>
    <View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderBottomColor": "#ededed",
                "borderBottomWidth": 1,
                "paddingBottom": 10,
                "paddingRight": 10,
                "paddingTop": 10,
              },
              undefined,
            ],
          ]
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
                "marginLeft": 10,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Object {
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <View
              style={undefined}
            >
              <View>
                <View
                  style={
                    Object {
                      "flexDirection": "row",
                      "justifyContent": "space-between",
                      "marginLeft": 10,
                    }
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "color": "darkslategrey",
                        "fontSize": 18,
                        "fontWeight": "bold",
                      }
                    }
                  >
                    description
                  </Text>
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "color": "midnightblue",
                        "fontSize": 18,
                      }
                    }
                  >
                    $ 
                    99
                  </Text>
                </View>
              </View>
            </View>
            <View
              style={undefined}
            >
              <View>
                <View
                  style={
                    Object {
                      "marginLeft": 10,
                    }
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "color": "grey",
                        "marginVertical": 2,
                      }
                    }
                  >
                    01/02/00  00:00
                  </Text>
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "color": "darkgrey",
                        "fontSize": 12,
                      }
                    }
                  >
                    Some long comment
                  </Text>
                </View>
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "alignItems": "flex-end",
                "flex": 0.15,
                "justifyContent": "center",
              }
            }
          >
            <View
              onLongPress={undefined}
              onPress={undefined}
              style={
                Array [
                  false,
                  false,
                  false,
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "justifyContent": "center",
                  },
                  undefined,
                ]
              }
              underlayColor="white"
            >
              <Text
                accessible={true}
                allowFontScaling={false}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "#bdc6cf",
                      "fontSize": 28,
                    },
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                      },
                      Array [
                        Object {},
                        undefined,
                      ],
                    ],
                    Object {
                      "fontFamily": "Material Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                  ]
                }
              >
                î—Œ
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "backgroundColor": "lightblue",
      }
    }
  >
    <View
      accessibilityComponentType="button"
      accessibilityLabel={undefined}
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#0C42FD",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Print
        </Text>
      </View>
    </View>
  </View>
</View>
`;
